zetasql/fuzzing/protobuf/zetasql_expression_extractor.cc

```diff
@@ -0,0 +1,144 @@
+  builder.clear();
+  return released;
+}
+
+void ProtoExprExtractor::Extract(const Expression& expr) {
+  using ExprType = Expression::ExprOneofCase;
+  if (expr.parenthesized()) {
+    Append("(");
+    if (expr.has_leading_pad()) {
+      Extract(expr.leading_pad());
+    }
+  }
+  
+  switch (expr.expr_oneof_case()) {
+    case ExprType::kLiteral:
+      Extract(expr.literal());
+      break;
+    case ExprType::kExpr:
+      Extract(expr.expr());
+      break;
+    default:
+      EXTRACT_DEFAULT(expr);
+      break;
+  }
+
+  if (expr.parenthesized()) {
+    if (expr.has_trailing_pad()) {
+      Extract(expr.trailing_pad());
+    }
+    Append(")");
+  }
+}
+
+void ProtoExprExtractor::Extract(const LiteralExpr& literal) {
+  using LitExprType = LiteralExpr::LiteralOneofCase;
+  switch (literal.literal_oneof_case()) {
+    case LitExprType::kSpecialLiteral:
+      switch (literal.special_literal()) {
+        case LiteralExpr::V_NULL:
+          return Append("NULL");
+        default:
+          Exit("Undefined Special Literal");
+      }
+    case LitExprType::kBoolLiteral:
+      return Append(literal.bool_literal() ? "TRUE" : "FALSE");
+    case LitExprType::kBytesLiteral:
+      return Append(literal.bytes_literal());
+    case LitExprType::kStringLiteral:
+      return Append(literal.string_literal());
+    case LitExprType::kIntegerLiteral:
+      return Extract(literal.integer_literal());
+    case LitExprType::kNumericLiteral:
+      return Extract(literal.numeric_literal());
+    default:
+      return EXTRACT_DEFAULT(literal);
+  }
+}
+
+void ProtoExprExtractor::Extract(const IntegerLiteral& integer) {
+  using IntergerType = IntegerLiteral::IntegerOneofCase;
+  switch (integer.integer_oneof_case()) {
+    case IntergerType::kInt32Literal:
+      return Append(integer.int32_literal());
+    case IntergerType::kUint32Literal:
+      return Append(integer.uint32_literal());
+    case IntergerType::kInt64Literal:
+      return Append(integer.int64_literal());
+    case IntergerType::kUint64Literal:
+      return Append(integer.uint64_literal());
+    default:
+      return EXTRACT_DEFAULT(integer);
+  }
+}
+
+void ProtoExprExtractor::Extract(const NumericLiteral& numeric) {
+  return Append(numeric.value());
+}
+
+void ProtoExprExtractor::Extract(const CompoundExpr& comp_expr) {
+  using CompoundExprType = CompoundExpr::CompoundOneofCase;
+  switch (comp_expr.compound_oneof_case()) {
+    case CompoundExprType::kBinaryOperation:
+      return Extract(comp_expr.binary_operation());
+    default:
+      return EXTRACT_DEFAULT(comp_expr);
+  }
+}
+
+void ProtoExprExtractor::Extract(const BinaryOperation& binary_operation) {
+  using zetasql_expression_grammar::BinaryOperation_Operator;
+  const static std::map<BinaryOperation_Operator, std::string> operators{
+      {BinaryOperation::PLUS, "+"},
+      {BinaryOperation::MINUS, "-"},
+      {BinaryOperation::MULTIPLY, "*"},
+      {BinaryOperation::DIVIDE, "/"},
+  };
```

Comment:
Why not use a switch-case statement here? You are creating a static red-black tree for the operator types, then doing a try-catch on the map's "at" method. A case statement seems much more explicit and probably optimizes better.